pacfetcher.go

```diff
@@ -80,6 +83,46 @@ func requireOK(resp *http.Response, err error) (*http.Response, error) {
 	}
 }
 
+// decodeDataURL decodes a data URL (e.g., data:text/plain;base64,SGVsbG8sIFdvcmxkIQ==).
+// It supports both base64 and URL-encoded data, and enforces the maxResponseBytes size limit.
+// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs for details.
+func decodeDataURL(uri string) ([]byte, error) {
+	parsed_url, err := url.Parse(uri)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing pac url: %w", err)
+	}
+
+	if parsed_url.Scheme != "data" {
+		return nil, nil
+	}
+	parts := strings.SplitN(parsed_url.Opaque, ",", 2)
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("Error parsing data URL: Invalid Format")
+	}
+
+	isBase64 := slices.Contains(strings.Split(parts[0], ";"), "base64")
+	if isBase64 {
```

Comment:
i think it would be nicer if you used [strings.Cut()](https://pkg.go.dev/strings#Cut) and [strings.HasSuffix()](https://pkg.go.dev/strings#HasSuffix) so we don't have to deal with slices and indexes, what do you think?

```suggestion
	metadata, data, ok := strings.Cut(parsedURL.Opaque, ",")
	if !ok {
		return nil, fmt.Errorf("Error parsing data URL: Invalid Format")
	}

    if strings.HasSuffix(metadata, ";base64") {  // ignore media type
```

if you use slices.Contains(), you might also run into an issue where the substring `base64` might appear on the wrong side of the `;`, and the behaviour would technically be incorrect